### PR TITLE
Add Redux Toolkit

### PR DESCRIPTION
**Pull request by SamTush**
## What does this PR do?

This PR is raised to implement the following features: 👍 :

- [x] Add redux Toolkit
- [x] Configure a Redux store 
- [x] Create slice for books
- [x] Create slice for categories

## Description of the task to be completed ↓↓ 

- Add Redux Toolkit

  
## How should it be tested

Clone the repository
```
git clone https://github.com/SamTush/bookstore.git
```
Navigate to the folder
```
cd bookstore
```
Checkout the branch
```
git checkout feature/redux-setup
```

## How should it be run

Run website in browser.
```
  npm start
```

### CODE REVIEW

File: src/redux/books/booksSlice.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+};
+
+export const booksSlice = createSlice({
+  name: 'books',
+  initialState,
+  reducers: {
+    addBook: () => {
+    },
+    removeBook: () => {
+    },
```
Comment (tugsem) [inline src/redux/books/booksSlice.js]: - [ ] Please complete these reducers for adding and removing a book from the state as the requirements asked.

**Review by tugsem — CHANGES_REQUESTED** [review]
##  STATUS: REQUIRED CHANGES ♻️
Hello @SamTush 👋 , 

Good job so far! 👏 
There are some issues that you still need to work on to go to the next project but you are almost there!

Alt Text
## Highlights 🚀 
✔️linters are passing
✔️correct use of gitflow
✔️professional documentation
## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/redux/books/booksSlice.js (hunk 1 above)
Comment (SamTush) [reply src/redux/books/booksSlice.js]: Thank you for reviewing my PR👍🏽. Cheer and happy coding😎

**Comment by SamTush** [comment]
Thank you for reviewing my PR @tugsem 👍🏽. Cheer and happy coding

**Review by julie-ify — APPROVED** [review]
Hi @SamTush,

## Highlights!
- Good code organization and working app ✔️
- Descriptive README ✔️ 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

# Status: Approved ✔️ 

Cheers and Happy coding!👏👏👏

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**